### PR TITLE
[19.01] Fix format_source for implicit conversion inputs and fasta.gz upload

### PR DESCRIPTION
**Pull request by mvdbeek**
While writing the test I also noticed that fasta.gz uploads were broken, so this also fixes that.
I also noticed that implicit conversion doesn't happen for collection inputs (they work fine if mapped over), but since we don't have any reports for that and it'll probably be a bigger change I'll reserve that for dev

Fixes https://github.com/galaxyproject/galaxy/issues/7622

### CODE REVIEW

File: test/functional/tools/sample_datatypes_conf.xml [hunk 1]
```diff
@@ -7,7 +7,7 @@
     <datatype extension="csv" type="galaxy.datatypes.tabular:CSV" display_in_upload="true" />
     <datatype extension="interval" type="galaxy.datatypes.interval:Interval" display_in_upload="true" description="File must start with definition line in the following format (columns may be in any order)." >
     </datatype>
-    <datatype extension="fasta" type="galaxy.datatypes.sequence:Fasta" auto_compressed_types="gz,bz2" display_in_upload="true">
```
Comment (martenson) [inline test/functional/tools/sample_datatypes_conf.xml]: Why is bz2 going away? And if so should we remove it for fastqsanger too?

Comment (martenson) [reply test/functional/tools/sample_datatypes_conf.xml]: oh it was restored later, nvm

File: lib/galaxy/model/mapping.py [hunk 2]
```diff
@@ -1926,6 +1926,9 @@ def simple_mapping(model, **kwds):
     implicitly_converted_datasets=relation(model.ImplicitlyConvertedDatasetAssociation,
         primaryjoin=(model.ImplicitlyConvertedDatasetAssociation.table.c.ldda_parent_id ==
                      model.LibraryDatasetDatasetAssociation.table.c.id)),
+    implicitly_converted_parent_datasets=relation(model.ImplicitlyConvertedDatasetAssociation,
+        primaryjoin=(model.ImplicitlyConvertedDatasetAssociation.table.c.ldda_id ==
+                     model.LibraryDatasetDatasetAssociation.table.c.id)),
```
Comment (nsoranzo) [inline lib/galaxy/model/mapping.py]: Alternatively you could add a backref at line 1608-1610.

Comment (mvdbeek) [reply lib/galaxy/model/mapping.py]: Alright, done that and also replaced `implicitly_converted_parent_datasets` for the HDA relation in the same way.

**Review by jmchilton — APPROVED** [review]

**Comment by martenson** [comment]
thanks @mvdbeek !

**Pull request merged**